#153756406 user can start game

sannytee opened the pull request:
#### What does this PR do?
Enables User to start or create game

#### Description of Task to be completed?
>- Implement modal
>- save game logs to database after game end

#### How should this be manually tested?
* from terminal, run: `git clone https://github.com/andela/kenya-cfh.git` to clone repository
* cd into the cloned repository and run: `npm install` to install all required dependencies
* create a `.env` file in the root directory and set up Mongodb configurations as described in the `.env.sample` file.
*  run `npm test` to ensure all tests are passing
* run `npm run start:dev` to startup application and open browser at localhost:3000. 
* create an account
* play game with friends or strangers

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#153756406
#### Screenshots (if appropriate)
<img width="1437" alt="screen shot 2018-01-21 at 4 41 00 pm" src="https://user-images.githubusercontent.com/28694656/35195868-0c249d52-feca-11e7-9316-ee93137ca76e.png">
<img width="1435" alt="screen shot 2018-01-21 at 4 40 34 pm" src="https://user-images.githubusercontent.com/28694656/35195869-0c48dbe0-feca-11e7-8669-65064ade54ef.png">

#### Questions:
N/A

sannytee committed:
feature(UI/UX): user can start game

[start #153756406]

>- update views files to include modal
>-  update css file
>- implement function to call modal

sannytee committed:
feature(database): save game log to database

[#153756406]

>- create game schema
>- implement game controller
>- add the route to save game log
>- implement middleware to verify user
>- update function to end game

sannytee committed:
feature(database): save game log to database

[#153756406]

>- create game schema
>- implement game controller
>- add the route to save game log
>- implement middleware to verify user
>- update function to end game

sannytee committed:
chore(test): write tests for starting game and saving game logs

[finishes #153756406]

temilaj commented on app/controllers/game.js:
```diff
@@ -0,0 +1,29 @@
+          gameLog: gameData
+        });
+      })
+      .catch(() => res.status(400).send({ message: 'Game logs not saved' }));
```
How are we sure this is a 400 statuscode and not a 500?

temilaj commented on config/middlewares/authorization.js:
```diff
@@ -1,33 +1,46 @@
  * Generic require login routing middleware
+ * @export
+ *
+ * @param {any} req request sent from the body
```
when writing JSDOC is great practice to include the appropriate data type and not `any`

temilaj commented on config/passport.js:
```diff
@@ -1,11 +1,12 @@
+import config from './config';
+
 var mongoose = require('mongoose'),
```
I see we have variables declared with the `var` keyword. We have also declared multiple variables in a single line. How can you improve on this?

temilaj commented on public/js/services/game.js:
```diff
@@ -197,6 +198,23 @@ angular.module('mean.system').factory('game', [
+            headers: { 'x-token': window.localStorage.token }
+          }
+        )
+          .then(response => response, error => error);
```
What happens here if an error occurs?

temilaj reviewed (changes requested):
Made some comments in this PR. However, there are some other comments I haven't made. We'll be talking about these during the TTL interaction session. I reckon you can find these errors if you take the time to go through this again, and ask a few peers to review

temilaj commented on config/express.js:
```diff
@@ -1,91 +1,88 @@
+
+    // express/mongo session storage
+    app.use(express.session({
+      secret: 'MEAN',
```
we should totally avoid directory including application settings or secret keys in the codebase

Billmike committed:
feature(chat feature):implement group chat feature

- setup chat html view
- integrate firebase and angular fire services
- setup client-side chat controller
- modify directives.js to include new chat view
- inject firebase module into app.js
- refactor app.js to use es6 syntax
- remove comments from chat.css file

[Finishes #153756414]

Victoradukwu committed:
feature(localisation): cards should be categorised by region
 - install node-localstorage package
 - add regionId to game socket, question and answer
 - the start button pops up a modal through which a player selects their region
 - modify the startGame function to select questions and answers based an inputed regionId
 [Finishes #153756413]

sannytee committed:
chore(eslint): fix eslint errors

sannytee committed:
chore(eslint): fix eslint errors
